addons/mail/static/src/owl/hooks/use_store_compare_keys.js

```diff
@@ -0,0 +1,121 @@
+odoo.define('mail.hooks.useStoreCompareKeys', function (require) {
+'use strict';
+
+const { useStore } = owl.hooks;
+
+/**
+ * Similar to useStore but to decide if a new render has to be done it compares
+ * the keys on the result, with an optional level of depth for each key, given
+ * as options `compareDepth`.
+ *
+ * It assumes that the result of the selector is always an object (or array).
+ *
+ * @param {function} selector @see useStore
+ * @param {object} [options={}] @see useStore
+ * @param {number|object} [options.compareDepth=0] the comparison depth, either
+ *  as number (applies to all keys) or as an object (depth for specific keys)
+ * @returns {Proxy} @see useStore
+ */
+function useStoreCompareKeys(selector, options = {}) {
+    const store = options.store || owl.Component.current.env.store;
+    const hashFn = store.observer.revNumber.bind(store.observer);
+    const isEqual = options.isEqual || ((a, b) => a === b);
+
+    /**
+     * Returns a function comparing whether two values are the same, which is just
+     * calling `isEqual` on primitive values and objects, but which also works for
+     * Owl Proxy in a temporal way: the current result of hashFn is compared to the
+     * previous result of hashFn (from the last time the function was called).
+     *
+     * It means that when this function is given Proxy the first call will always
+     * return false, and consecutive calls will not lead to the same result:
+     * it returns true the first time after a change happended inside the Proxy, and
+     * then always returns false until a new change is made.
+     *
+     * @returns {function} function taking two arguments, and comparing them as
+     *  explained above.
+     */
+    function proxyComparator() {
```

Comment:
I would keep function outside, but not exported

Reply:
not changing, using closure for hashfn & isequal